### PR TITLE
Upgrade to Python 3.7

### PR DESCRIPTION
**Pull request by alex9smith**
The router uses Python to generate nginx config files on startup, but Python 2.7 was EOL on January 1st 2020 and needs upgrading.

Install Python 3.7 instead of 2.7, update dependencies and add `python3` everywhere a python script is called. There is more
cleanup work that can be done in this app (see [Trello](https://trello.com/c/kQ6vKiRy/1162-3-manage-router-docker-image-in-docker-base-repo-split-me) for more details), but that's out of scope for this PR.

Note that this Docker image is hard to test. I have run the unit tests and successfully built and run the image locally. I've also logged into the container and checked that the environment variables have been injected into the `.conf` files everywhere they're referenced.

https://trello.com/c/hwA5XVWT/1497-upgrade-the-router-off-python-27

### CODE REVIEW

File: nginx.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-export DM_CLOUDFRONT_IPS=$(/app/scripts/get-cloudfront-ips.py)
+export DM_CLOUDFRONT_IPS=$(python3 /app/scripts/get-cloudfront-ips.py)
```
Comment (lfdebrux) [inline nginx.sh:3]: Rather than doing this we should change the hashbang line in the script file to call `python3` explicitly

**Comment by lfdebrux** [comment]
@alex9smith The easiest way to test this is to push an image to Docker Hub and use that image for the router app in the preview space

File: nginx.sh (hunk 1 above)
Comment (alex9smith) [reply nginx.sh:3]: Good point. I will do that

**Comment by alex9smith** [comment]
> If we are upgrading is there a reason not to go all the way to Python 3.9?

Yes - Debian buster (which `nginx:1.19.5` uses) only has up to 3.7 in the [package library](https://packages.debian.org/buster/python3) since it's the [officially supported 3.X version](https://wiki.debian.org/Python) . We'd need to compile from source to get a higher version.

> The easiest way to test this is to push an image to Docker Hub and use that image for the router app in the preview space

Thanks, I'll give that a go

**Review by lfdebrux — APPROVED** [review]

**Comment by alex9smith** [comment]
I released the new image to preview and got passes from the [functional](https://ci.marketplace.team/view/Functional,%20visual,%20and%20smoke%20tests/job/functional-tests-preview/23041/) and [visual regression](https://ci.marketplace.team/job/visual-regression-preview/3015/) tests. All working as expected 👍

**Pull request merged**